### ISSUE TITLE
view scripts not executing in bootstrap\modal for the first time

### ISSUE THREAD

**Issue by buttflattery** [yiisoft/yii2#16867]
If you are using `bootstrap\Modal` to load a form inside the modal window, and you show the modal window on the click of a button, the scripts are not executed the first modal window loads the view either they are inline using `registerJs('alert("hello")')` or registered via `Asset::register($this)`, but if you close the modal window and click the button again then the scripts are executed as soon as the modal window opens and loads the view. Also the statement `View::POS_READY` has no effect when used with `$this->registerJs('alert("hello")',\yii\web\View::POS_READY);`, it is not wrapped inside the `$(document).ready()` block.

### What steps will reproduce the problem?
Create a view `views/site/modal.php`  and add the modal code like below with an extra button and `js` code to open the modal window on button click and bind event `shown.bs.modal` to load the view inside the modal content section once modal is shown

```
<?php

use yii\bootstrap\Modal;
use yii\helpers\Html;

Modal::begin([
    'id' => 'contact',
]);
?>

<div class="modal-header">
    <h4>Contact</h4>
</div>

<div class="modal-body">

</div>

<div class="modal-footer">
    <?php
    echo Html::submitButton('Add Folder Now', ['class' => 'btn btn-primary']);
    ?>

</div>

<?php Modal::end() ;

 echo Html::button('button', ['class'=>'btn btn-primary','id'=>'add-folder-button']);
 
 $js=<<< js
        $("#add-folder-button").on('click',function(){
            $( '#contact' ).modal( {show:true});
            $( '#contact' ).appendTo("body");
        });
         
         $( '#contact' ).on( 'shown.bs.modal', function ( event ) {
            // var button = $(event.relatedTarget); // Button that triggered the modal
            let modal = $( this );
            modal.find( '.modal-header > h4' ).text( 'Add Folder' );
            modal.find( '.modal-body' ).load( 'contact');
        } );
js;
$this->registerJs($js, \yii\web\View::POS_READY);
```

Create an action inside the `SiteController` to load the above view

```
public function actionModal(){
        return $this->render('modal');
    }
```
_____________________________

Now modify the `actionContact`  inside the `SiteController` like below to be loaded inside the modal window

```
/**
     * Displays contact page.
     *
     * @return Response|string
     */
    public function actionContact()
    {
        $model = new ContactForm();
        if ($model->load(Yii::$app->request->post()) && $model->contact(Yii::$app->params['adminEmail'])) {
            Yii::$app->session->setFlash('contactFormSubmitted');
            return $this->refresh();
        }
        return $this->renderAjax('contact', [
            'model' => $model,
        ]);
    }
```

For the view, you can use the default `contact.php` view inside the `views/site` folder just add the below code on top of the view 
```

$js=<<< js
        alert("hello");
js;
$this->registerJs($js, \yii\web\View::POS_READY);
```

### What is the expected result?
on clicking the button the modal window should load and as soon the view is loaded it should show you the `alert("hello")` , and if I cycle through the fields it should run the `ActiveForm` client validation. 

_Note : i even tried providing `'enableClientValidation'=>true` in the form options._

### What do you get instead?
it Does nothing neither it shows the alert neither it runs the validation, instead if you close and then click the button again to open the modal window then everything works correctly. the Alert is shown and the client validation works. but if you refresh the page again it wont work again until you open the modal window once and close it and open it again.

The above scenario works perfectly in Yii 1.x versions and i have it working i am working on a yii1.x extension to transform it to yii2.x

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      | 5.6
| Operating system | windows

**Comment by buttflattery** [comment]
@samdark sorry if this is being considered as a question as I did not intend to, the minimum code to regenerate the issue is already added, and it is the most basic way that is being used widely around for opening a view inside the modal window, but it simply does not work. I would take this question to StackOverflow although just to make sure there is something I am doing wrong. But can you just confirm if this works accurately in your case?

**Comment by samdark** [comment]
No, it won't work. POS_READY is fired once on loading a document. Loading model contents is not that.

**Comment by buttflattery** [comment]
@samdark i don't think so you are following me on this, i don't care if the `POS_READY` works or not, lets say there isn't any inline javascript code that is written using `$this->registerJs()` in the view that you are loading inside the modal, its just an `ActiveForm`. Now, it should trigger the validation as soon as you TAB through the fields. 

If you agree on this prove me wrong when i say 

**"The ActiveForm validation will not work on the first modal load, BUT the second time, if you dont refresh the page and just close the modal window and load it again."**

Why does it work on the second time then, either it should work on the first time or it should never work. And according to you the inline scripts also shouldnt work why does it displays the alert the second time modal window is loaded without refreshing the page.

**Comment by GHopperMSK** [comment]
I have reproduced the situation and got the same result. I don't know why, but renaming `modal-body` class name solve it.

```
Modal::begin([
    'id' => 'contact',
]);
?>

    <div class="modal-header">
        <h4>Contact</h4>
    </div>

    <div class="modal-bodyy">

    </div>

    <div class="modal-footer">
        <?php
            echo Html::submitButton('Add Folder Now', ['class' => 'btn btn-primary']);
        ?>

    </div>

<?php Modal::end() ;

echo Html::button('Show', ['class'=>'btn btn-primary','id'=>'add-folder-button']);

$js = <<< js
    $("#add-folder-button").on('click',function(){
        $( '#contact' ).modal({show: true});
        $( '#contact' ).appendTo("body"); // do you really need this?!
    });

    $( '#contact' ).on( 'shown.bs.modal', function ( event ) {
        let modal = $( this );
        modal.find( '.modal-header > h4' ).text( 'Add Folder' );
        modal.find( '.modal-bodyy' ).load( 'contact'); 
    });
js;

$this->registerJs($js, \yii\web\View::POS_READY);
```
User case:
1. refresh the page
2. press Show button
3. got Modal window
4. `Contact` changes to `Add Folder`
5. Alert "hello" appeared
6. press Ok
7. Contact view loaded to Div with modal-bodyy class name
8. got validation errors after press Submit button

**Comment by samdark** [comment]
That means a handler is bound to `.modal-body` that results in this behavior.

**Comment by GHopperMSK** [comment]
here is my code:
```
    $( '#contact' ).on( 'shown.bs.modal', function ( event ) {
        let modal = $( this );
        let body = modal.find( '.modal-body' );
        console.dir(body);
        // if (body.length > 1) {
        //     body.splice(0,1);
        // }
        body.load( 'contact');
    });
```
On first button press I got:
![image](https://user-images.githubusercontent.com/19670887/51062956-e7bdc080-1609-11e9-8030-6fe87e6d3e0e.png)

On second:
![image](https://user-images.githubusercontent.com/19670887/51062981-06bc5280-160a-11e9-9b0d-c046a2af7c77.png)

**Comment by samdark** [comment]
Weird... no idea why there are two modal bodies...

**Comment by GHopperMSK** [comment]
I think they are widget itself, and its content. 
The code
```
Modal::begin([
    'id' => 'contact',
]);
...
Modal::end() ;
``` gives us one container. Than, content
```
<div class="modal-header">
    <h4>Contact</h4>
</div>
```
gives us another one. That is it.

**Comment by terabytesoftw** [comment]
Closing this as not a Yii2 core bug.

`yii\bootstrap\Modal` already renders its own modal body wrapper. Its `bodyOptions` default to `['class' => 'modal-body']`. The reproduction adds another `.modal-body` inside the widget-generated body, so `modal.find('.modal-body')` matches multiple nested elements and `.load()` is applied to the wrong/nested nodes. That explains why the first load overwrites the modal content and later attempts appear to work differently.

The `POS_READY` behavior in an AJAX-rendered response is also expected: in AJAX mode Yii renders `POS_READY` / `POS_LOAD` JavaScript as normal script blocks at the end of the response instead of wrapping them in `jQuery(document).ready()`.

The correct approach is to use the Modal widget options such as `header`, `footer`, and `bodyOptions`, or target a unique placeholder inside the generated modal body rather than selecting `.modal-body` broadly.
